### PR TITLE
ci: migrate to Github Actions

### PR DESCRIPTION
**Pull request by delta1**
## Description

- Switch CI to Github Actions
  1. CI workflow runs cargo fmt check and clippy, build on ubuntu/macos/windows, and test on ubuntu simultaneously [example](https://github.com/delta1/tari/actions/runs/1030198337)
  2. Test RFC docs on any PR [example](https://github.com/delta1/tari/runs/3066309197?check_suite_focus=true), and deploy RFC docs to gh-pages on any push to development branch (same as circle ci did) [example](https://github.com/delta1/tari/runs/3066309272?check_suite_focus=true) / [rendered](https://delta1.github.io/tari/)
  3. Security audit - unchanged
  4. Binary builds - unchanged 

- Integration tests remain on Circle CI for now, there is a WIP version here that needs debugging 

## Motivation and Context

Consolidating CI with one provider as much as possible 

## How Has This Been Tested?

Github actions

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I have squashed my commits into a single commit.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -231,25 +231,25 @@ workflows:
...
+      #       branches:
+      #         ignore: gh-pages
+      # - test-tari-release:
+      #     filters:
+      #       branches:
+      #         ignore: gh-pages
+      # - clippy:
+      #     filters:
+      #       branches:
+      #         ignore: gh-pages
```
Comment (SWvheerden) [inline .circleci/config.yml:255]: Why are these commented out?
Rather delete them or leave a comment as to why they are commented out.

**Review by SWvheerden — APPROVED** [review]
LGTM, just one nit

File: .circleci/config.yml (hunk 1 above)
Comment (delta1) [reply .circleci/config.yml:255]: Yeah happy to delete them but @mikethetike mentioned just commenting them out.

**Comment by delta1** [comment]
Looks like the required status checks for the protected branch will need to be changed in the repo settings 

image

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -40,24 +67,14 @@ jobs:
...
-        with:
-          SQLITE3_LIB_DIR: 'C:\vcpkg\installed\x64-windows\lib'
-          OPENSSL_DIR: 'C:\Program Files\OpenSSL-Win64'
-          LIBCLANG_PATH: 'C:\Program Files\LLVM\bin'
-
-      # this is a hack to fix an issue with building libclang in MSVC
-      # it should be fixed in release 16.9 of MSVC
-      # issue https://github.com/microsoft/STL/issues/1300
-      # temp fix https://github.com/mono/CppSharp/pull/1514/files
-      - name: fix intrin.h file
```
Comment (StriderDM) [inline .github/workflows/ci.yml:57]: Might want to leave this in, just commented out, in the event it is a problem again at some point in future. Remember that quite a bit of time was spent on fixing this when it was a problem.

Comment (delta1) [reply .github/workflows/ci.yml:57]: It was fixed upstream

Comment (delta1) [reply .github/workflows/ci.yml:57]: Fixed in VS 16.9 - https://github.com/microsoft/STL/issues/1300#issuecomment-789542493 

windows-2019 runners are already on VS 16.10 - https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#visual-studio-enterprise-2019

**Review by stringhandler — APPROVED** [review]

**Pull request merged**